components/Feature/SharePlan/index.js

```diff
@@ -73,13 +84,44 @@ const SharePlan = ({ error, plan, customerUrl, onPlanShared }) => {
               className={css['share-plan__collaborators-list']}
               data-testid="share-by-sms-row-test"
             >
-              <Checkbox
-                name="share-by-sms"
-                label={getNumber(plan.numbers[0])}
-                value={getNumber(plan.numbers[0])}
-                disabled={!plan.numbers[0]}
-                onClick={handleSelectNumber}
-              />
+              {editNumber && (
+                <input
+                  onChange={handleNumberChange}
+                  value={numberText}
+                  autoFocus
+                  data-testid="edit-number-input-test"
+                />
+              )}
+
+              {editNumber && (
+                <button
+                  onClick={saveNumber}
+                  className="govuk-details__summary-text"
+                  data-testid="save-number-button-test"
+                >
+                  Save
+                </button>
+              )}
```

Comment:
Maybe we can use govuk text inputs/buttons or reuse components if we already have something?